src/main/kotlin/com/github/michaelbull/result/bind/ResultBindingImpl.kt

```diff
@@ -0,0 +1,49 @@
+package com.github.michaelbull.result.bind
+
+import com.github.michaelbull.result.Err
+import com.github.michaelbull.result.Ok
+import com.github.michaelbull.result.Result
+import com.github.michaelbull.result.getOrElse
+import com.github.michaelbull.result.bind.NoStackTraceException.BindFailure
+
+/**
+ * Calls the specified function [block] with [ResultBindingImpl] as its receiver and returns a [Result] of either [Ok] with value [V] or [Err] with error [E].
+ *
+ * [ResultBinding.bind] extension function is made available to any [Result] object inside this [block].
+ * [ResultBinding.bind] will attempt to unwrap the [Result] to return its [Ok] value locally.
+ * If the result is a [Err], then the [binding] block terminates with that [ResultBinding.bind] and returns a [Result] containing the error of the failed binding.
+ *
+ * This allows for easy unwrapping of result type objects along a happy path.
+ * for examples, see tests
+ * @sample com.github.michaelbull.result.bind.ResultBindingTest
+ *
+ */
+inline fun <V, E> binding(crossinline block: ResultBinding<E>.() -> V): Result<V, E> {
```

Comment:
I think we should make this function name more specific, for example `resultBinding` or `bindResult`, as `binding` feels too generic. Let me know if you have a good proposal for this.

Reply:
Upon further inspection, it looks like a library called [`bow`](https://bow-swift.io/docs/patterns/monad-comprehensions/) for swift calls this function "binding", so let's keep it that way.

Reply:
ah nice, yep bow is the counter part to arrow so this makes sense!

Reply:
but i agree, perhaps it is a bit too generic, and I agree `resultBinding` might be the best way to phrase what is is - my only issue is it becomes quite a long word to use for what is essentially a keyword.

Reply:
interestingly if you go far back enough in the arrow repo (back to when it was called kategory) they had considered calling `bind` -> `await`. but might have been a good choice to not confuse this as being something async or with coroutines